core/src/main/scala/kafka/log/ProducerStateManager.scala

```diff
@@ -158,14 +163,14 @@ private[log] class ProducerAppendInfo(val producerId: Long,
                          producerEpoch: Short,
                          offset: Long,
                          timestamp: Long): CompletedTxn = {
-    if (this.producerEpoch > producerEpoch)
+    if (isProducerFenced(producerEpoch))
       throw new ProducerFencedException(s"Invalid producer epoch: $producerEpoch (zombie): ${this.producerEpoch} (current)")
 
     if (this.coordinatorEpoch > endTxnMarker.coordinatorEpoch)
       throw new TransactionCoordinatorFencedException(s"Invalid coordinator epoch: ${endTxnMarker.coordinatorEpoch} " +
         s"(zombie), $coordinatorEpoch (current)")
 
-    if (producerEpoch > this.producerEpoch) {
+    if (producerEpoch != this.producerEpoch) {
```

Comment:
No, but the check needed to be updated due to the wraparound logic. Since we have already verified that the producer is not a zombie in the `isProducerFenced` check above, a producer epoch which does not match the current epoch must be the new epoch.